openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/reader/traffic/TrafficLinkMetadata.java

```diff
@@ -0,0 +1,87 @@
+
+    private TrafficEnums.LinkTravelDirection travelDirection;
+    private TrafficEnums.FunctionalClass functionalClass = TrafficEnums.FunctionalClass.CLASS5;
+    private TrafficEnums.NoYesEnum frontageRoad = TrafficEnums.NoYesEnum.NO;
+    private TrafficEnums.NoYesEnum ramp = TrafficEnums.NoYesEnum.NO;
+    private TrafficEnums.NoYesEnum roundabout = TrafficEnums.NoYesEnum.NO;
+    private TrafficEnums.FerryType ferryType = TrafficEnums.FerryType.NO;
+    private TrafficEnums.NoYesEnum specialTrafficFigure = TrafficEnums.NoYesEnum.NO;
+
+    public TrafficLinkMetadata(Collection<Property> properties) {
+        for (Property property : properties) {
+            try {
+                String propertyName = property.getName().toString();
+                switch (propertyName) {
+                    case "DIR_TRAVEL":
+                        travelDirection = TrafficEnums.LinkTravelDirection.forValue(property.getValue().toString());
+                        break;
+                    case "FUNC_CLASS":
+                        functionalClass = TrafficEnums.FunctionalClass.forValue(Integer.parseInt(property.getValue().toString()));
+                        break;
+                    case "FRONTAGE":
+                        frontageRoad = TrafficEnums.NoYesEnum.forValue(property.getValue().toString().charAt(0));
+                        break;
+                    case "RAMP":
+                        ramp = TrafficEnums.NoYesEnum.forValue(property.getValue().toString().charAt(0));
+                        break;
+                    case "ROUNDABOUT":
+                        roundabout = TrafficEnums.NoYesEnum.forValue(property.getValue().toString().charAt(0));
+                        break;
+                    case "FERRY_TYPE":
+                        ferryType = TrafficEnums.FerryType.forValue(property.getValue().toString().charAt(0));
+                        break;
+                    case "SPECTRFIG":
+                        specialTrafficFigure = TrafficEnums.NoYesEnum.forValue(property.getValue().toString().charAt(0));
+                        break;
+                    default:
+                        break;
+                }
+            } catch (Exception e) {
+                LOGGER.debug("Couldn't process property.");
+            }
+        }
+    }
+
+    public boolean isFerry() {
+        return ferryType != TrafficEnums.FerryType.NO;
+    }
+
+
+    public boolean isRoundAbout() {
+        return roundabout != TrafficEnums.NoYesEnum.NO;
+    }
+
+    public TrafficEnums.LinkTravelDirection getTravelDirection() {
+        return this.travelDirection;
+    }
+
+    public TrafficEnums.FunctionalClass functionalClass() {
+        return functionalClass;
+    }
+
+    public int getFunctionalClassWithRamp() {
+        if (this.ramp == TrafficEnums.NoYesEnum.NO) {
+            return this.functionalClass.getFunctionalClass();
+        } else if (this.functionalClass == TrafficEnums.FunctionalClass.CLASS1) {
```

Comment:
Use switch in stead of an if-else-chain to make the code more readable.

Reply:
@MichaelsJP